src/NuGet.Core/NuGet.Build.Tasks/RestoreTaskEx.cs

```diff
@@ -59,6 +59,11 @@ public sealed class RestoreTaskEx : Task, ICancelableTask, IDisposable
         /// </summary>
         public bool Interactive { get; set; }
 
+        /// <summary>
+        /// Gets a value indicating whether or not <see cref="SolutionPath" /> contains a value.
+        /// </summary>
+        public bool IsSolutionPathDefined => !string.IsNullOrWhiteSpace(SolutionPath) && !string.Equals(SolutionPath, "*Undefined*", StringComparison.OrdinalIgnoreCase);
```

Comment:
Can be `internal`?

Reply:
This might as well be public since its only looking at another public property